### DISCUSSION TITLE
Can I connect a database to the JSON Feed for instant updates?

### DISCUSSION THREAD

**Discussion by nataliefaith2** [code4recovery/12-step-meeting-list#1148] [Q&A] [answered]
Is there a way to connect a database (Access/ Excel) to the Meeting Plugin/ JSON Feed so when the different areas make updates to meeting times/locations the webiste feed can be updated?

I saw this page: https://sheets.code4recovery.org/

And I feel like that's close to this result but it's an import as oppose to a constant feed, is that correct?

My team would love this solution.

**Comment by brianw-area24** [comment]
There are 2 ways to import meetings from another source, but are on the "Import & Export" page. You are correct, both these methods are imports, rather than automatically importing on periodic basis

The first method is a one-time import of a CSV file you upload. I see this a great method to do the initial import of meetings into the plug. 

The other way is to import JSON Feed(s). Currently it does require you to click the `Refresh` button each time you want to import the feed. There is a Change Detection Feature to help you know when a feed should be refreshed. I think a member has been working on a feature that will allow feeds to be refreshed automatically, but that's not available at this time.

If I understand what you want to do, I'd say your best option would be to:

- Put your spreadsheet on Google Sheets
- Put your Google sheet into https://sheets.code4recovery.org
- Put the JSON Feed into the Import & Export page, using the change detection feature

Brian W.
Code 4 Recovery volunteer

**Reply by conrac01** [reply]
A member of our area tech committee mentioned his district used the plugin WPCrontrol to auto refresh the meeting updates from the area's master JSON file.  
It's configured as:
'Hook'  tsml_scan_data_source
'Argument' ["http://area_feed.com/district?districtNum=4"]
Action' tsml_scan_data_source()

But it does not seem to work for me.

I also tried to use the plugin "Get URLCron"
https://district4.areaname.org/wp-admin/edit.php?post_type=tsml_meeting&page=import

When I refresh on the Import & Export tab I receive the following message after pressing "refresh"

"The District 4 data source's next scheduled run is June 18, 2023 11:02 am. You can adjust the recurrences and the times that the job ('tsml_scan_data_source') runs with the WP_Crontrol plugin."

Could the WPCrontrol refresh be working and just not be reflected in the "Last Refresh" date/timne on the Import & Export tab?

Thanks,  Ed M.

**Reply by kiyote33** [reply, marked as answer]
Hi @conrac01, the quick answer is "No, the refresh does not work that way!" 
The behavior described by the member of your area tech committee could currently only be achieved by using custom code to overwrite the default behavior of our plugin **tsml_scan_data_source** function code.  Our code is designed to simply detect data source import meetings which have been updated since the last refresh timestamp, and then list the results of the scan in a generated email which is sent to the email addresses listed on the Settings page under the **Change Notification Emails** heading. A recipient of the generated email (titled "Data Source Changes Detected: <data_source_name>) can then click on the embedded link in the email which takes them to their WordPress _Meetings "Import & Export" admin page_ where pressing the <data_source_name> **Refresh** button resyncs  their website meeting list with those meetings found in the import feed.  
I would suggest contacting the member of your tech committee for details on exactly what they did to achieve the auto refresh feature, something which is not provided by our plugin at this time.
 
Hope this answers your question...
